process.c

```diff
@@ -3020,77 +3005,67 @@ NORETURN(static VALUE f_exec(int c, const VALUE *a, VALUE _));
- *
- *  Replaces the current process by running the given external _command_, which
- *  can take one of the following forms:
+ *    exec([env, ] command_line, options = {})
+ *    exec([env, ] exe_path, options  = {})
+ *    exec([env, ] exe_path, *args, options  = {})
  *
- *  [<code>exec(commandline)</code>]
- *	command line string which is passed to the standard shell
- *  [<code>exec(cmdname, arg1, ...)</code>]
- *	command name and one or more arguments (no shell)
- *  [<code>exec([cmdname, argv0], arg1, ...)</code>]
- *	command name, +argv[0]+ and zero or more arguments (no shell)
+ *  Replaces the current process by doing one of the following:
  *
- *  In the first form, the string is taken as a command line that is subject to
- *  shell expansion before being executed.
+ *  - Passing string +command_line+ to the shell.
+ *  - Invoking the executable at +exe_path+.
  *
- *  The standard shell always means <code>"/bin/sh"</code> on Unix-like systems,
- *  otherwise, <code>ENV["RUBYSHELL"]</code> or <code>ENV["COMSPEC"]</code> on
- *  Windows and similar.  The command is passed as an argument to the
- *  <code>"-c"</code> switch to the shell, except in the case of +COMSPEC+.
+ *  The new process is created using the
+ *  {exec system call}[https://www.man7.org/linux/man-pages/man1/exec.1p.html];
+ *  it may inherit some of its environment from the calling program
+ *  (possibly including open file descriptors).
  *
- *  If the string from the first form (<code>exec("command")</code>) follows
- *  these simple rules:
+ *  Argument +env+, if given, is a hash that affects +ENV+ for the new process;
+ *  see {Execution Environment}[rdoc-ref:Process@Execution+Environment].
  *
- *  * no meta characters,
- *  * not starting with shell reserved word or special built-in,
+ *  Argument +options+ is a hash of options for the new process;
+ *  see {Execution Options}[rdoc-ref:Process@Execution+Options].
  *
- *  Ruby invokes the command directly without shell.
+ *  The first required argument is one of the following:
  *
- *  You can force shell invocation by adding ";" to the string (because ";" is
- *  a meta character).
+ *  - +command_line+ if it is a string,
+ *    and if it begins with a shell reserved word or special built-in,
+ *    or if it contains one or more metacharacters.
+ *  - +exe_path+ otherwise.
  *
- *  Note that this behavior is observable by pid obtained
- *  (return value of spawn() and IO#pid for IO.popen) is the pid of the invoked
- *  command, not shell.
+ *  <b>Argument +command_line+</b>
  *
- *  In the second form (<code>exec("command1", "arg1", ...)</code>), the first
- *  is taken as a command name and the rest are passed as parameters to command
- *  with no shell expansion.
+ *  \String argument +command_line+ is a command line to be passed to a shell;
+ *  it must begin with a shell reserved word or special built-in,
+ *  and may also contain arguments and options for that command.
```

Comment:
```suggestion
 *  it must begin with a shell reserved word, begin with a special built-in,
 *  or contain meta characters.
 *  It may also contain arguments and options for that command.
```